worker.go

```diff
@@ -343,5 +352,14 @@ func go_frankenphp_finish_request(threadIndex C.uintptr_t, isWorkerRequest bool)
 
 	if isWorkerRequest {
 		thread.Unpin()
+		workers[fc.scriptFilename].threadMutex.Lock()
+		defer workers[fc.scriptFilename].threadMutex.Unlock()
+		for i, t := range workers[fc.scriptFilename].threads {
+			if t == thread {
+				// remove thread from worker threads
+				workers[fc.scriptFilename].threads = append(workers[fc.scriptFilename].threads[:i], workers[fc.scriptFilename].threads[i+1:]...)
+				break
+			}
+		}
```

Comment:
I probably doesn't make sense to remove the thread from the worker after each request. Did you mean to put this logic at the end of the worker script?

Leaving this here will break `worker.handleRequest` since you're destroying the thread's request chan after each request.

Reply:
In my tests, this is only reached when the worker request (ie, the worker script) ends. We're only removing the thread from the list of active threads because it will not be reused (and in fact, "lost") after this function ends. This is the last chance to remove it from the slice.

Reply:
This function is called after each worker request (`frankenphp_handle_request`), not at the end of the worker script. You can confirm this by just adding a log here `logger.Warn("worker request over")`

Reply:
I see. Hmmm, I'll have to dig into it a bit deeper. Right now, there is a memory leak:

![image](https://github.com/user-attachments/assets/cce9cd0e-b9fd-4419-9c66-b6e4edde90cb)

As you can see, the number of threads that belong to a worker grow unbounded, despite there only supposed to be a single thread for this worker (see: `num = 1`)

Reply:
In all honesty, I don't believe this can be fixed until #1137.